other-docs/guides/code-review/standards.md

```diff
@@ -1,52 +1,206 @@
+Escaping is used to ensure that data is safe to be output to the browser. Data must be escaped using one of Altis' built-in escaping functions before output. The type of escaping function to use depends on the context in which the data is output.
+
+This is because data might be safe or unsafe depending on where it is output. Data that is perfectly safe to output between two HTML tags might not be safe to output inside of a piece of inline JavaScript.
+
+When writing code, always escape immediately before output. This is referred to as **late escaping**. This makes it clear when and how data is escaped, making the code easy to review and to understand.
+
+It also avoids introducing security issues by accident. If the contents of a variable are escaped first, and then output later in the code, then this code is secure at the time it's written. However if at a later time, then escaping is removed from the variable, then all the instances in which the variable is output are now vulnerable. Late escaping avoids this problem.
+
+
+### Validate and sanitize input {#validate-sanitize-input}
+
+*Sniff: `HM.Security.ValidatedSanitizedInput`*
+
+All user input must be validated and sanitized before being used in the codebase.
+
+Validation and sanitisation are two separate but related concepts.
+
+When validating data, you are looking for certain criteria in the data. Or simply put, you’re saying “I want the data to have this, this, and this”. Sanitisation on the other hand is about removing all the harmful elements from the data. In essence you’re saying “I don’t want the data to have this, this, and this”.
+
+But the difference is more than just conceptual. With validation, we store the data once we have verified it’s valid. If not, we discard it.
+
+With sanitisation, we take the data, and remove everything we don’t want. This means that we might change the data during the sanitisation process. So in the case of user input, it is not guaranteed that all the input is kept. So it’s important that you choose the right sanitisation functions, to keep the data intact.
+
+
+## Check CSRF tokens (nonces) on all requests {#nonce}
+
+*Sniff: `HM.Security.NonceVerification`*
+
+Altis provides a system called [nonces](https://developer.wordpress.org/plugins/security/nonces/) which prevent [cross-site request forgery (CSRF)](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html) vulnerabilities.
+
+Nonces should be checked on every request which performs a destructive action, as they check the user intended to take the action.
+
+
+## Hide debugging data
+
+### Do not expose filenames {#no-expose-files}
+
+*Sniff: `WordPress.Security.PluginMenuSlug`*
+
+`__FILE__` should not be used in admin menu slugs, as it exposes information to users about the underlying filesystem. While this is not a security issue directly, it can make issue escalation much easier, so is to be avoided as part of a security-in-depth mentality.
```

Comment:
```suggestion
`__FILE__` should not be used in admin menu slugs, as it exposes information to users about the underlying filesystem. While this is not a security issue directly, it can make privilege escalation much easier, so is to be avoided as part of a security-in-depth mentality.
```